osfoffline/filesystem_manager/sync_local_filesystem_and_db.py

```diff
@@ -11,209 +13,185 @@
     DirCreatedEvent,
 )
 from watchdog.observers import Observer
+from sqlalchemy.orm.exc import MultipleResultsFound, NoResultFound
+from osfoffline.database_manager.db import session
+from osfoffline.polling_osf_manager.osf_query import OSFQuery
 from osfoffline.database_manager.models import User, Node, File, Base
 from osfoffline.exceptions.item_exceptions import InvalidItemType, FolderNotInFileSystem
 from osfoffline.exceptions.local_db_sync_exceptions import LocalDBBothNone, IncorrectLocalDBMatch
 from osfoffline.utils.path import ProperPath
+from osfoffline.polling_osf_manager.api_url_builder import api_url_for, NODES, USERS
+
+
+logger = logging.getLogger(__name__)
+
+
+def get_sha256(path, chunk_size=1024**2):
+    s = hashlib.sha256()
+    with open(path, 'rb') as fobj:
+        while True:
+            chunk = fobj.read(chunk_size)
+            if not chunk:
+                break
+            s.update(chunk)
+    return s.hexdigest()
 
 
 class LocalDBSync(object):
     COMPONENTS_FOLDER_NAME = 'Components'
 
-    def __init__(self, absolute_osf_dir_path, observer, user):
-        if not isinstance(observer, Observer):
-            raise TypeError
-        if not isinstance(user, User):
-            raise TypeError
-        if not os.path.isdir(absolute_osf_dir_path):
+    def __init__(self, observer, user, loop=None):
+        self.user = user
+        self.observer = observer
+        self.loop = loop or asyncio.get_event_loop()
+        self.osf_query = OSFQuery(self.loop, self.user.oauth_token)
+        self.osf_folder = self.user.osf_local_folder_path
+
+        if not os.path.isdir(self.osf_folder):
             raise FolderNotInFileSystem
 
-        self.osf_path = ProperPath(absolute_osf_dir_path, True)
-        self.observer = observer
-        self.user = user
+        self.osf_path = ProperPath(self.osf_folder, True)
 
-    def emit_new_events(self):
-        for local_child, db_child in self._make_local_db_tuple_list(self.osf_path, self.user):
-            self._emit_single_event_and_recurse(local_child, db_child)
+    def sync_db(self):
+        logger.info('Beginning initial sync')
+        events, nodes = [], [
+            node for node in
+            self.user.top_level_nodes
+            if node.osf_id in self.user.guid_for_top_level_nodes_to_sync
+        ]
+        for node in nodes:
+            logger.info('Resyncing node {}'.format(node))
+            events.extend(self.loop.run_until_complete(self.crawl_node(node)))
```

Comment:
Note to self: check this during functional testing step.

Watchdog events appear to be just data objects- although these objects are created (in `infer_event`), I'm not clear on whether the events are actually being fired in a way that lets `osf_event_handler` pick up on them.

In the old version we [added them](https://github.com/CenterForOpenScience/OSF-Offline/pull/97/files#diff-ad501704142723e44bdb2e2d5267671eL171) to an event queue on the observer, but this appears to no longer be the case, and the value of `events` does not appear to be used (or returned) by the `sync_db` function.